apps/prairielearn/src/middlewares/selectAndAuthzInstructorQuestion.js

```diff
@@ -29,7 +34,10 @@ module.exports = function (req, res, next) {
     };
     sqldb.queryZeroOrOneRow(sql.select_and_auth, params, function (err, result) {
       if (ERR(err, next)) return;
-      if (result.rowCount === 0) return next(error.make(403, 'Access denied'));
+      let row = result.rows[0];
+      if (row.question.course_id !== res.locals.course.id && !row.shared_with_course) {
```

Comment:
This will error if `result` returns zero rows (e.g. if the user tries to view a question that doesn't exist). Aside: this is why our new `queryOptionalRow` + TypeScript is so good, it would force you to handle this situation!

Rather than needing separate code paths for "does not exist" and "is not shared", consider shifting the check into the query, something roughly like `WHERE ... AND (question.course_id = course.id OR is_shared_with_course)`.

Reply:
That's a great point about too many code paths here, I've moved the check into the query to simplify. 

I'll definitely remember that about `queryOptionalRow` for new code, I'm going to go ahead and avoid refactoring this file for now though.